packages/utils/src/deepClone.ts

```diff
@@ -0,0 +1,15 @@
+export const deepClone = <T>(value: T): T => {
+    if (value === undefined) {
+        return value;
+    }
+
+    let clone;
+
+    if (typeof structuredClone === 'function') {
+        clone = structuredClone(value);
+    } else {
+        clone = JSON.parse(JSON.stringify(value));
```

Comment:
We should probably use some better polyfill than this. There are already available.

Reply:
Never heard of it before, core-js has a polyfill. (I know we are cautious about included new dependencies, but just sharing).

https://github.com/zloirock/core-js#structuredclone